File: testdata/fixtures/javascript/CWE-532.js
// CWE-532: Insertion of Sensitive Information into Log File
//
// Console and logger calls that write a password, secret or token expose it
// in log files which are often world-readable or shipped to third parties.

function leak(password, secret, apiKey) {
  // ruleid: js-sensitive-log
  console.log(password);
  // ruleid: js-sensitive-log
  console.info("password=" + password);
  // ruleid: js-sensitive-log
  console.debug(secret);
}

function safe() {
  // ok: js-sensitive-log
  console.log("user logged in");
  // ok: js-sensitive-log
  console.info("request processed");
  // ok: js-sensitive-log
  console.debug("redacted");
  // ok: js-sensitive-log
  logger.info("done");
}

File: testdata/fixtures/rust/CWE-532.rs
// CWE-532: Sensitive info into log — password reaches log::info! with taint.
use log::{info, debug};

fn leak(password: &str, secret: &str, api_key: &str) {
    // ruleid: rust-log
    log::info!("password={}", password);
    // ruleid: rust-log
    info!("secret {}", secret);
    // ruleid: rust-log
    log::debug!("api_key={}", api_key);
    // ruleid: rust-log
    println!("token {}", secret);
}

fn safe() {
    // ok: rust-log
    log::info!("user logged in");
    // ok: rust-log
    info!("request processed");
    // ok: rust-log
    println!("redacted");
}
